datafusion/src/physical_plan/hash_join.rs

```diff
@@ -1025,14 +1053,42 @@ impl Stream for HashJoinStream {
                     );
                     self.num_input_batches += 1;
                     self.num_input_rows += batch.num_rows();
-                    if let Ok(ref batch) = result {
+                    if let Ok((ref batch, ref left_side)) = result {
                         self.join_time += start.elapsed().as_millis() as usize;
                         self.num_output_batches += 1;
                         self.num_output_rows += batch.num_rows();
+
+                        if self.join_type == JoinType::Left {
```

Comment:
nit, I think it would be better to do an explicit match on the join type enum here to exhaust all cases. that way, when we add outer join in the future, the compiler will remind us to update the check here.

Reply:
Thanks @houqp I agree, updated the code!